src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -390,6 +418,43 @@ def build_cmd(self, env, host, **kwargs):
 
         return tst_cmd
 
+    def convert_xml(self, xml_file):
+        """Modify the xml file"""
+
+        with open(xml_file, 'r') as fd:
+           with open('{}.xml'.format(xml_file), 'w') as ofd:
+              for line in fd:
+                  if self.src_dir in line:
+                      L = re.sub(r'<dir>\/*' + self.src_dir + '\/*',
+                                 r'<dir>',
+                                 line)
+                      ofd.write(L)
+                  else:
+                      ofd.write(line)
+              os.unlink(xml_file)
+
+        return 0
+
+    def convert_xml_files(self):
+        """Check valgrind memcheck log files for errors."""
+
+        daos_test_shared_dir = os.getenv('DAOS_TEST_SHARED_DIR',
+                                         os.getenv('HOME'))
+
+        self.log.info("Parsing log path %s", daos_test_shared_dir)
+        if not os.path.exists(daos_test_shared_dir):
+            self.log.info("Path does not exist")
+            return 1
+
+        xml_filename_fmt = r"^valgrind\.\S+\.memcheck$"
+        memcheck_files = list(filter(lambda x: re.match(xml_filename_fmt, x),
+                                os.listdir(daos_test_shared_dir)))
+
+        for filename in memcheck_files:
+            self.convert_xml(daos_test_shared_dir + "/" + filename)
+
+         return 0
```

Comment:
(pylint-syntax-error) unindent does not match any outer indentation level (<unknown>, line 456)